src/Microsoft.AspNet.Mvc.Razor/Precompilation/RazorErrorExtensions.cs

```diff
@@ -20,15 +22,19 @@ public static Diagnostic ToDiagnostics([NotNull] this RazorError error, [NotNull
                 defaultSeverity: DiagnosticSeverity.Error,
                 isEnabledByDefault: true);
 
-            var textSpan = new TextSpan(error.Location.AbsoluteIndex, error.Length);
-            var linePositionStart = new LinePosition(error.Location.LineIndex, error.Location.CharacterIndex);
-            var linePositionEnd = new LinePosition(error.Location.LineIndex,
-                                                   error.Location.CharacterIndex + error.Length);
-            var linePositionSpan = new LinePositionSpan(linePositionStart, linePositionEnd);
-
-            var location = Location.Create(filePath, textSpan, linePositionSpan);
+            var location = error.Location;
+            if (location == SourceLocation.Undefined)
+            {
+                location = SourceLocation.Zero;
+            }
```

Comment:
this seems like a localized fix for a more general problem.  seem to see `Undefined` locations most often in cases where the error is in another file.  so we need to pass more information along and emit diagnostics showing the actual locations, not the start of the current file.